### DISCUSSION TITLE
new release for gguf package

### DISCUSSION THREAD

**Discussion by jaideepr97** [ggml-org/llama.cpp#10782] [General]
Hi all - correct me if I'm wrong, but I believe the [gguf pypi package](https://pypi.org/project/gguf/) is published out of https://github.com/ggerganov/llama.cpp/tree/master/gguf-py/gguf
The latest release (v0.10.0) was published in August 2024

I have noticed that there have been a lot of additions to that package since the last release, including recognition for new model architectures like granite
I am in need of consuming these new changes and wanted to see if I could request a new release for the GGUF package? 
(cc @gabe-l-hart )

On a separate note, I am also interested in consuming the `convert_hf_to_gguf.py` script in my code, but have not found a good way to import it since it is not published anywhere (to my knowledge), requiring me to manually copy the script and make it fit my needs
If anyone has ideas on better ways to consume that, I'd be interested as well
Thanks!

**Comment by ggerganov** [comment]
I've bumped the version: https://github.com/ggerganov/llama.cpp/pull/10788
Please ping me if there is no new package published. I forgot if I have to manually tag the release, or if some CI workflow will automatically do it.

> On a separate note, I am also interested in consuming the convert_hf_to_gguf.py script in my code, but have not found a good way to import it since it is not published anywhere (to my knowledge), requiring me to manually copy the script and make it fit my needs
If anyone has ideas on better ways to consume that, I'd be interested as well

For now, you should manually synchronize it. We don't have plans to publish it since it would be too difficult to maintain backwards compat.

**Reply by jaideepr97** [reply]
ack, thanks for the quick response!

**Reply by jaideepr97** [reply]
@ggerganov looks like pypi is still pointing to 0.10.0

**Reply by ggerganov** [reply]
Yes, I was thinking that the workflow will start automatically, but it's not the case. Just launched it manually: https://github.com/ggerganov/llama.cpp/actions/runs/12300483697. Let's see if it works.

**Reply by jaideepr97** [reply]
its published now, thanks!

**Reply by gabe-l-hart** [reply]
Thanks for the quick turn around!
